### PR TITLE
Update config files for Solr & Fedora

### PR DESCRIPTION
**Pull request by mark-dce**
Update config files for easier compatability with solr_wrapper and
fcrepo_wrapper.

Also adds sample ldap config file.

### CODE REVIEW

**Comment by mark-dce** [comment]
@no-reply I think this gets us to a working development environment.  I'm working on test and CI now.

**Comment by no-reply** [comment]
This replaces #30

File: .fcrepo_wrapper.sample [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Place any default configuration for solr_wrapper here
```
Comment (no-reply) [inline .fcrepo_wrapper.sample]: `fcrepo_wrapper` instead of `solr_wrapper`?

File: config/solr.yml.sample [hunk 2]
```diff
@@ -1,10 +1,8 @@
-# This is a sample config file that does not have multiple solr instances. You will also need to be sure to
-# edit the fedora.yml file to match the solr URL for active-fedora. 
 development:
-  url: http://localhost:8983/solr/development
+  url: <%= ENV['SOLR_URL'] || "http://127.0.0.1:#{ENV.fetch('SOLR_DEVELOPMENT_PORT', 8983)}/solr/development" %>
 test: &TEST
-  url: <%= "http://127.0.0.1:#{ENV['TEST_JETTY_PORT'] || 8983}/solr/test" %>
+  url: http://localhost:<%= ENV.fetch('SOLR_TEST_PORT', 8985) %>/solr/test
```
Comment (no-reply) [inline config/solr.yml.sample]: could this just stay on 8983? It wouldn't be possible to run both development and test servers under that configuration, but the test suite should be able to connect.

File: config/fedora.yml.sample [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
   password: fedoraAdmin
-  url: http://127.0.0.1:8983/fedora
-test: &TEST  
+  url: http://127.0.0.1:<%= ENV['FCREPO_DEVELOPMENT_PORT'] || 8984 %>/rest
+  base_path: /dev
+test:
   user: fedoraAdmin
   password: fedoraAdmin
-  url: <%= "http://127.0.0.1:#{ENV['TEST_JETTY_PORT'] || 8983}/fedora-test" %>
+  url: http://127.0.0.1:<%= ENV['FCREPO_TEST_PORT'] || 8986 %>/rest
```
Comment (no-reply) [inline config/fedora.yml.sample]: Similar to https://github.com/curationexperts/MIRA/pull/31/files#r122817724, could/should this just be `8984` for now?

**Comment by no-reply** [comment]
Pending comments, this looks good. As it stands, it doesn't allow connections on `bundle exec rake`. With suggested changes, I switch from connection errors to:

```  
RSolr::Error::Http - 404 Not Found
  Error:     Not Found

  URI: http://localhost:8983/solr/test/select?wt=json&sort=system_create_dtsi+asc&fl=id&q=*%3A*&qt=standard&start=0&rows=1000
```

File: config/blacklight.yml.sample [hunk 4]
```diff
@@ -0,0 +1,9 @@
+development:
+  adapter: solr
+  url: <%= ENV['SOLR_URL'] || "http://127.0.0.1:#{ENV.fetch('SOLR_DEVELOPMENT_PORT', 8983)}/solr/development" %>
```
Comment (no-reply) [inline config/blacklight.yml.sample]: `solr/hydra-development`?

File: config/blacklight.yml.sample [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+test: &test
+  adapter: solr
+  url: http://localhost:<%= ENV.fetch('SOLR_TEST_PORT', 8985) %>/solr/test
```
Comment (no-reply) [inline config/blacklight.yml.sample]: `solr/hydra-test`?

File: config/blacklight.yml.sample (hunk 4 above)
Comment (mark-dce) [inline config/blacklight.yml.sample]: The core names are really arbitrary and can either be hydra-*environment* or just *environment*. I was going for the shorter, but am happy with either.  As long as the usage is consistent through the wrapper config files, service config files, and ci task, everything should work fine.  Let me know which feels better `hydra-development` or just `development`

File: config/blacklight.yml.sample (hunk 5 above)
Comment (mark-dce) [inline config/blacklight.yml.sample]: see discussion for `solr\hydra-development`

**Comment by mark-dce** [comment]
I think the main problem with having both `test` and `development` instances on the same ports is that you need the `ci` task to start a server when it's running under travis.  So CI generally launches a solr and fedora instance.  If your running them on the same ports as your development services, you need to stop the development services before you run ci.  

I think there has to be a simpler way to run solr and fedora instances that:
* support a persistent development data store
* support a transient test data store (empty each run)
* support running single tests during TDD and diagnostic cycles
* support launching the instances in a CI context

The current code looks like it does most of that except for trying to run `rake ci` when you already have test instances of solr and fedora running (eg. for TDD).  If you want to keep test servers running in your local environment, you need to use `rake spec` instead of `rake ci`.

If that's OK, this code should be ready to merge or I can squash it all into a single commit first if that seems better.

**Pull request merged**